app/controllers/projects_controller.rb

```diff
@@ -123,6 +123,13 @@ def update
       }).run()
 
       redirect_to edit_project_path(@project, anchor: params[:anchor] || 'home')
+    elsif params[:cancel_project] == 'true'
+      api = ApiWrapper.new current_user
+      api.request("rpc/cancel_project", {
+        body: { _project_id: @project.id }.to_json,
+        action: :post
+      }).run()
```

Comment:
currently we dont have habilities to call direct refund from API, so maybe we can use something like:
```ruby
@project.refund_all_payments if @project.reload && @project.failed?
```
we also can notify users instead create a contribution_notification directly on RPC abstracting the refund and notify_users method on project_observer